src/main/java/org/opensearch/sdk/ExtensionsRunner.java

```diff
@@ -105,6 +111,23 @@ PluginResponse handlePluginsRequest(PluginRequest pluginRequest) {
         return pluginResponse;
     }
 
+    /**
+     * Handles a request from OpenSearch and invokes the extension point API corresponding with the request type
+     *
+     * @param request  The request to handle.
+     * @return A response to OpenSearch for the corresponding API
+     */
+    TransportResponse handleOpenSearchRequest(OpenSearchRequest request) {
+        // Read enum
+        switch (request.getRequestType()) {
+            case REQUEST_OPENSEARCH_NAMED_WRITEABLE_REGISTRY:
+                return namedWriteableRegistryApi.handleNamedWriteableRegistryRequest(request);
+            // Add additional request handlers here
+            default:
+                return null;
```

Comment:
Instead of null let's return an exception here?

Reply:
This is following the same pattern as the handleExtensionRequest in ExtensionOrchestrator.